blackjack.py

```diff
@@ -79,23 +79,32 @@ def print_entire_hand(self):
         for card in self.hand:
             print card
 
+    def check_natural_win(self):
+        if len(self.hand) == 2:
+            if self.hand[0].value == 'Ace' and self.hand[1].value in ['Jack', 'Queen', 'King']:
+                return True
+            elif self.hand[0].value in ['Jack', 'Queen', 'King'] and self.hand[1].value == 'Ace':
+                return True
+            else:
+                return False
+        return False
+
     def get_soft_hand_value(self):
         """
         Calculates hand value with Ace as 1 (Soft hand value)
         :return: int
         """
         soft_hand_value = 0
-        for card in self.hand:
-            if self.hand[0].value == 'Ace' and self.hand[1].value in ['Jack', 'Queen', 'King']:
-                soft_hand_value = 21.1
-            elif self.hand[0].value in ['Jack', 'Queen', 'King'] and self.hand[1].value == 'Ace':
-                soft_hand_value = 21.1
-            elif card.value.isdigit():
-                soft_hand_value += int(card.value)
-            elif card.value == 'Ace':
-                soft_hand_value += 1
-            else:
-                soft_hand_value += 10
+        if self.check_natural_win():
+            soft_hand_value = 21.1
+        else:
+            for card in self.hand:
+                if card.value.isdigit():
+                    soft_hand_value += int(card.value)
+                elif card.value == 'Ace':
+                    soft_hand_value += 1
+                else:
+                    soft_hand_value += 10
```

Comment:
This would be easier if you added a `points` attribute to the `Card` class, you'd be able to replace the above 7 lines with something like this:
```
soft_hand_value += sum(card.points for card in self.hand)
```

Going further, your whole method could be (assuming the `points` attribute of the Ace card is 1 point):
```
    def get_soft_hand_value(self):
        return (21.1 if self.check_natural_win()
                else sum(card.points for card in self.hand))
```

And then you could do this:
```
    def get_hard_hand_value(self):
        return (self.get_soft_hand_value() +
                10 * sum(card.value == 'Ace' for card in self.hand))
```

Or you could even do this:
```
    def get_hand_value(self, hard=False):
        return (21.1 if self.check_natural_win()
                else sum(card.points for card in self.hand) +
                hard * 10 * sum(card.value == 'Ace' for card in self.hand))
```

Reply:
@romainrichard 
First of all, this is beautiful. Thank you! 
Jesse also suggested implementing a points system, but seeing it work like this really shows me how much better that would be. 
I'm gonna try implement this today 🙂 

I think there is one thing I don't understand, in the last example you wrote. 


I'm not sure I get 
```
sum(card.points for card in self.hand) +
                hard * 10 * sum(card.value == 'Ace' for card in self.hand))
```

hard is set as 'False', then you're multiplying False by 10 for each ace in the hand?

Reply:
Ah yes, it's one of those things that makes Python fun.

Here's a clue:
```
In [1]: 10 * True
Out[1]: 10

In [2]: 10 * False
Out[2]: 0
```

Reply:
Oh my god, The only thing I could think of was that's what It did. But I've NEVER seen that before, nor did I think it would work. Holy crap. 

https://www.reactiongifs.us/wp-content/uploads/2014/03/stop_penis_erect_archer.gif